Toolchain/BuildIt.sh

```diff
@@ -91,6 +91,52 @@ buildstep() {
     "$@" 2>&1 | sed $'s|^|\x1b[34m['"${NAME}"$']\x1b[39m |'
 }
 
+# === DEPENDENCIES ===
+buildstep dependencies echo "Checking whether 'make' is available..."
+if ! command -v ${MAKE:-make} >/dev/null; then
+    buildstep dependencies echo "Please make sure to install GNU Make (for the '${MAKE:-make}' tool)."
+    exit 1
+fi
+
+buildstep dependencies echo "Checking whether 'patch' is available..."
+if ! command -v patch >/dev/null; then
+    buildstep dependencies echo "Please make sure to install GNU patch (for the 'patch' tool)."
+    exit 1
+fi
+
+buildstep dependencies echo "Checking whether 'makeinfo' is available..."
+if ! command -v makeinfo >/dev/null; then
+    buildstep dependencies echo "Please make sure to install GNU Texinfo (for the 'makeinfo' tool)."
+    exit 1
+fi
+
+buildstep dependencies echo "Checking whether your C compiler works..."
+cat >test.c <<'PROGRAM'
+int main() {}
+PROGRAM
+if ! ${CC:-cc} -o test test.c >/dev/null; then
+    buildstep dependencies echo "Please make sure to install a working C compiler."
+    rm -f test.c
+    exit 1
+fi
+rm -f test test.c
+
+if [ "$SYSTEM_NAME" != "Darwin" ]; then
+    for lib in gmp mpc mpfr; do
+        buildstep dependencies echo "Checking whether the $lib library and headers are available..."
+        cat >test.c <<PROGRAM
```

Comment:
out of curiosity, is there a difference between this and `<<'PROGRAM'` above?

Reply:
Yeah, actually there is. This one interpolates variables (`$lib`) while the other doesn't. :)